.github/workflows/main.yml

```diff
@@ -33,7 +33,7 @@ permissions:
 
 jobs:
   pre-commit:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```

Comment:
Could you please check if pre-commit required python3.7?

Reply:
Updated to use latest python (3.12) with the ubuntu-latest